### PR TITLE
AUTO-9017: polish chain modules

### PR DESCRIPTION
**Pull request by FelixFan1992**

### CODE REVIEW

File: contracts/src/v0.8/automation/dev/chains/ChainModuleBase.sol [hunk 1]
```diff
@@ -4,6 +4,8 @@ pragma solidity 0.8.19;
 import {IChainModule} from "../interfaces/v2_2/IChainModule.sol";
 
 contract ChainModuleBase is IChainModule {
+  uint256 private constant FIXED_GAS_OVERHEAD = 30000;
```
[the remark this answers is not in the document]
Comment (FelixFan1992) [reply contracts/src/v0.8/automation/dev/chains/ChainModuleBase.sol]: it turns out block.number only costs 12 gas and blockhash costs a few hundreds. so 500 to cap it off.

File: contracts/src/v0.8/automation/dev/chains/ScrollModule.sol [hunk 2]
```diff
@@ -5,18 +5,18 @@ import {IScrollL1GasPriceOracle} from "../../../vendor/@scroll-tech/contracts/sr
...
-  /// @dev SCROLL_ORACLE_ADDR is the address of the L1GasPriceOracle precompile on Optimism.
+    hex"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
+  /// @dev SCROLL_ORACLE_ADDR is the address of the ScrollL1GasPriceOracle precompile on Scroll.
   /// @dev reference: https://docs.scroll.io/en/developers/transaction-fees-on-scroll/#estimating-the-l1-data-fee
   address private constant SCROLL_ORACLE_ADDR = 0x5300000000000000000000000000000000000002;
   IScrollL1GasPriceOracle private constant SCROLL_ORACLE = IScrollL1GasPriceOracle(SCROLL_ORACLE_ADDR);
 
-  uint256 private constant FIXED_GAS_OVERHEAD = 20000;
-  uint256 private constant PER_CALLDATA_BYTE_GAS_OVERHEAD = 20;
+  uint256 private constant FIXED_GAS_OVERHEAD = 15000;
```
Comment (infiloop2) [inline contracts/src/v0.8/automation/dev/chains/ScrollModule.sol:18]: Should this be 45000? (cost with 0x calldata)

File: contracts/src/v0.8/automation/dev/chains/OptimismModule.sol [hunk 3]
```diff
@@ -5,16 +5,16 @@ import {OVM_GasPriceOracle} from "../../../vendor/@eth-optimism/contracts/v0.8.9
...
-    hex"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
+    hex"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
   /// @dev OVM_GASPRICEORACLE_ADDR is the address of the OVM_GasPriceOracle precompile on Optimism.
   /// @dev reference: https://community.optimism.io/docs/developers/build/transaction-fees/#estimating-the-l1-data-fee
   address private constant OVM_GASPRICEORACLE_ADDR = 0x420000000000000000000000000000000000000F;
   OVM_GasPriceOracle private constant OVM_GASPRICEORACLE = OVM_GasPriceOracle(OVM_GASPRICEORACLE_ADDR);
 
-  uint256 private constant FIXED_GAS_OVERHEAD = 20000;
-  uint256 private constant PER_CALLDATA_BYTE_GAS_OVERHEAD = 20;
+  uint256 private constant FIXED_GAS_OVERHEAD = 30000;
```
Comment (infiloop2) [inline contracts/src/v0.8/automation/dev/chains/OptimismModule.sol:16]: Should this be 60000?

File: contracts/src/v0.8/automation/dev/chains/ArbitrumModule.sol [hunk 4]
```diff
@@ -16,8 +16,7 @@ contract ArbitrumModule is ChainModuleBase {
   address private constant ARB_GAS_ADDR = 0x000000000000000000000000000000000000006C;
   ArbGasInfo private constant ARB_GAS = ArbGasInfo(ARB_GAS_ADDR);
 
-  uint256 private constant FIXED_GAS_OVERHEAD = 20000;
-  uint256 private constant PER_CALLDATA_BYTE_GAS_OVERHEAD = 20;
+  uint256 private constant FIXED_GAS_OVERHEAD = 5000;
```
Comment (infiloop2) [inline contracts/src/v0.8/automation/dev/chains/ArbitrumModule.sol:19]: Is this finalized?

File: contracts/src/v0.8/automation/dev/chains/OptimismModule.sol (hunk 3 above)
Comment (FelixFan1992) [reply contracts/src/v0.8/automation/dev/chains/OptimismModule.sol:16]: yes the gas cost calculated by g1 - g2 is about 60K

File: contracts/src/v0.8/automation/dev/chains/ScrollModule.sol (hunk 2 above)
Comment (FelixFan1992) [reply contracts/src/v0.8/automation/dev/chains/ScrollModule.sol:18]: the gas cost calculated by g1 - g2 is about 45K

File: contracts/src/v0.8/automation/dev/chains/ArbitrumModule.sol (hunk 4 above)
Comment (FelixFan1992) [reply contracts/src/v0.8/automation/dev/chains/ArbitrumModule.sol:19]: yes this includes the gas usage for 3 calls combined:
`ARB_SYS.arbBlockNumber();`
`ARB_SYS.arbBlockHash(n);` and
`ARB_GAS.getCurrentTxL1GasFees();`
each takes about 1200 gas based on g1 - g2 formula.
and provide a little buffer.

**Pull request merged**